The 1.85 MSRV is inaccurate with all features

The 1.85 MSRV is inaccurate with all features

```
error[E0658]: `let` expressions in this position are unstable
  --> /home/runner/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/object_store-0.14.1/src/client/token.rs:81:12
   |
81 |         if let Some(cache) = self.cache.read().await.as_ref()
   |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: see issue #53667 <https://github.com/rust-lang/rust/issues/53667> for more information

error[E0658]: `let` expressions in this position are unstable
  --> /home/runner/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/object_store-0.14.1/src/client/token.rs:88:12
   |
88 |         if let Some(cache) = guard.as_ref()
   |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: see issue #53667 <https://github.com/rust-lang/rust/issues/53667> for more information

For more information about this error, try `rustc --explain E0658`.
error: could not compile `object_store` (lib) due to 2 previous errors
```

See https://github.com/zarrs/zarrs_object_store/actions/runs/29831899978/job/88638534711

The CI action that calls cargo-msrv could append `--all-features` to catch this.

--- a/.github/workflows/rust.yml
+++ b/.github/workflows/rust.yml
@@ -55,4 +55,4 @@ jobs:
       - name: Check
         run: |
           # run `cargo msrv verify` to see problems
-          cargo msrv verify --output-format=json || exit 1
+          cargo msrv verify --all-features --output-format=json || exit 1

--- a/src/client/token.rs
+++ b/src/client/token.rs
@@ -78,18 +78,18 @@ impl<T: Clone + Send + Sync> TokenCache<T> {
             })
         };
 
-        if let Some(cache) = self.cache.read().await.as_ref()
-            && is_token_valid(cache)
-        {
-            return Ok(cache.token.token.clone());
+        if let Some(cache) = self.cache.read().await.as_ref() {
+            if is_token_valid(cache) {
+                return Ok(cache.token.token.clone());
+            }
         }
 
         let mut guard = self.cache.write().await;
-        if let Some(cache) = guard.as_ref()
-            && is_token_valid(cache)
-        {
-            // Refresh race
-            return Ok(cache.token.token.clone());
+        if let Some(cache) = guard.as_ref() {
+            if is_token_valid(cache) {
+                // Refresh race
+                return Ok(cache.token.token.clone());
+            }
         }
 
         let cached = f().await?;

